src/runtime/future/future.h

```diff
@@ -79,4 +121,37 @@ void future_register_callback(pony_ctx_t **ctx,
  * puts on hold the processing of this message.
  */
 void future_await(pony_ctx_t **ctx, future_t *fut);
+
+extern pony_type_t future_type;
+
+void future_discharge(pony_ctx_t **ctx, future_t *fut);
+
+/* VANILLA FUTURE API */
```

Comment:
It would be great if the definition of *vanilla* futures et al is specified in the header files (or somewhere).
From the PR description:
>future.c : futures with multiple get, await and chain operations.
vanilla_future.c : futures with only one get operation.
poly_vanilla_future.c : futures with multiple get operations.

it's not clear to me if a vanilla future has a single get operation or if it means that I cannot get multiple times from it (which could happen if there is an optimisation to reuse the future, e.g.)